lib/src/view/board_editor/board_editor_screen.dart

```diff
@@ -0,0 +1,378 @@
+
+                final defaultBoardSize = constraints.biggest.shortestSide;
+                final isTablet = isTabletOrLarger(context);
+                final remainingHeight =
+                    constraints.maxHeight - defaultBoardSize;
+                final isSmallScreen =
+                    remainingHeight < kSmallRemainingHeightLeftBoardThreshold;
+                final boardSize = isTablet || isSmallScreen
+                    ? defaultBoardSize - kTabletBoardTableSidePadding * 2
+                    : defaultBoardSize;
+
+                final direction =
+                    aspectRatio > 1 ? Axis.horizontal : Axis.vertical;
+
+                return Flex(
+                  direction: direction,
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  mainAxisSize: MainAxisSize.max,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    _PieceMenu(
+                      boardSize,
+                      direction: flipAxis(direction),
+                      side: boardEditorState.orientation.opposite,
+                      isTablet: isTablet,
+                    ),
+                    _BoardEditor(
+                      boardSize,
+                      orientation: boardEditorState.orientation,
+                      isTablet: isTablet,
+                      pieces: boardEditorState.pieces.unlock,
+                    ),
+                    _PieceMenu(
+                      boardSize,
+                      direction: flipAxis(direction),
+                      side: boardEditorState.orientation,
+                      isTablet: isTablet,
+                    ),
+                  ],
+                );
+              },
+            ),
+          ),
+        ),
+        const _BottomBar(),
+      ],
+    );
+  }
+}
+
+class _BoardEditor extends ConsumerWidget {
+  const _BoardEditor(
+    this.boardSize, {
+    required this.isTablet,
+    required this.orientation,
+    required this.pieces,
+  });
+
+  final double boardSize;
+  final bool isTablet;
+  final Side orientation;
+  final Pieces pieces;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final boardPrefs = ref.watch(boardPreferencesProvider);
+
+    return ChessboardEditor(
+      size: boardSize,
+      pieces: pieces,
+      orientation: orientation,
+      settings: ChessboardEditorSettings(
+        pieceAssets: boardPrefs.pieceSet.assets,
+        colorScheme: boardPrefs.boardTheme.colors,
+        enableCoordinates: boardPrefs.coordinates,
+        borderRadius: isTablet
+            ? const BorderRadius.all(Radius.circular(4.0))
+            : BorderRadius.zero,
+        boxShadow: isTablet ? boardShadows : const <BoxShadow>[],
+      ),
+      pointerMode: ref.watch(boardEditorControllerProvider).editorPointerMode,
+      onDiscardedPiece:
+          ref.read(boardEditorControllerProvider.notifier).discardPiece,
```

Comment:
One should not use `ref.read` in `build()` methods. Also it is better to put it at the beginning of `build()` so that we know the provider dependencies of a widget.